builds/azure-pipelines/build-release.yml

```diff
@@ -23,8 +23,39 @@ stages:
   displayName: 'Release Build and Publish'
 
   jobs:
-  - job: BuildTestPublish
-    displayName: 'Release: Build, Test and Publish on Linux'
+  - job: BuildTest
+    displayName: 'Build and Test'
+
+  # Verifies cross-platform build and test
+    strategy:
+      matrix:
+        mac:
+          imageName: 'macos-latest'
+          # Disabling integration tests on macOS due to issues with running MSSQL on Docker
+          # We need to set up a self-hosted agent with Docker running by default: https://github.com/microsoft/azure-pipelines-tasks/issues/12823
+          testFilter: '--filter FullyQualifiedName!~Integration'
+          testServer: ''
+        windows:
+          imageName: 'windows-latest'
+          testFilter: ''
+          testServer: '(LocalDb)\MSSQLLocalDB'
+
+    pool:
+      vmImage: '$(imageName)'
+
+    workspace:
+      clean: all
+
+    steps:
+      - template: 'template-steps-build-test.yml'
+        parameters:
+          solution: '$(solution)'
+          configuration: '$(configuration)'
+          nugetVersion: '$(nugetVersion)'
+          testFilter: '$(testFilter)'
+
+  - job: BuildTestPublishLinux
```

Comment:
I'd just combine this into the job above and then have the publish step only run on linux

Reply:
I tried to do that but couldn't figure out how to run the publish template only for linux. I tried adding
`condition: eq(variables['Agent.OS'], 'linux')`
and
`${{ if eq($(Agent.OS), 'linux') }}:`
but both did not work.

Reply:
Hm, odd. Ok well go ahead and merge this in now then. It's not a huge deal and we can investigate later if needed